### PR TITLE
[build] Add requirements.txt

### PR DESCRIPTION
**Pull request by mapellidario**
As discussed privately with @amaltaro and in WMCore dev weekly meetings, I open this PR to provide T0 with its own `requirements.txt` file, based on its spec file.

Details of the spec file:

* used spec file `t0.spec`
* from the repo [cms-sw/cmsdist, branch comp_gcc630](https://github.com/cms-sw/cmsdist/tree/comp_gcc630)
* the current last commit in this repo is [cms-sw/cmsdist, commit 9e502ba](https://github.com/cms-sw/cmsdist/tree/9e502ba45b7dd50315872c72fa2b146013d811b2)

The `requirements.txt` file is build with

* [mapellidario/cmssw-dist-parse, tag 0.1](https://github.com/mapellidario/cmssw-dist-parse/releases/tag/0.1)
* with the command `python3 parse_spec.py -d /path/to/github.com/cmsdist/ -f t0.spec --no-recursive`
* with some manual changes to `cheetah`, `cx-oracle` and `mysqldb-python` package names

### CODE REVIEW

**Comment by germanfgv** [comment]
Hi @mapellidario. Thank you for providending the file. Could you please clarify how will this affect the release building process for us at Tier0? 

We are about to create a new tag based in WMCore 1.4.1.patch4. Usually we would modify t0.spec to update WMCore version and requirements.

**Comment by mapellidario** [comment]
I would leave any definitive comment to @amaltaro, but as far as I have understood your current build process is not going to change any time soon. 
This will be used in the future (I am not able to provide a timescale) if and when the build process will be based more on pip and less on spec files.

Please Alan, correct me if I am wrong.

**Comment by amaltaro** [comment]
That's correct, it does not change anything in the build process.
This requirements.txt file is important because it provides a list of dependencies which T0 depends on. Be that a human or a machine in the future.
In addition to that, you get automated GH alerts when any of those software + version has a public security issue. With that said, it's very important that we keep it up-to-date (especially important to me, because if we change dependencies for WMAgent, it will likely affect T0 too, so I need to make sure to feed that information back here).

In short, it can be merged at any time it's convenient to you.
Thanks for putting this PR in, Dario.
